### DISCUSSION TITLE
Can you have a job "fail" but not fail the workflow?

### DISCUSSION THREAD

**Discussion by gilzow** [community/community#56441] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a workflow that runs after a pull request is accepted into the default branch.
Job1 checks out the repository and evaluates the recent push to see if it came from a certain branch name.
What I would like is for the remaining jobs (job2, job3, job4) to only run if Job1 determines the push did come from that certain branch. If the push isn't from that branch, they should not run/be skipped. 

I know about job dependency but those seem to all require that job1 _succeed_.  However, I don't want Job1 to report a fail because then that marks the workflow as failed even though it isn't a true "Failure". 

Is there a way to have subsequent jobs depend on an earlier job but not have the earlier job fail? OR have it fail but somehow not have the workflow itself marked as "failed"?

**Comment by airtower-luna** [comment, marked as answer]
> Is there a way to have subsequent jobs depend on an earlier job but not have the earlier job fail?

Yes. The natural thing to do would be to have your job1 [set a job output](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idoutputs) that tells the other jobs if they should run (instead of failing), and then add [`if` conditions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif) using that output on the other jobs.

**Reply by gilzow** [reply]
thanks. what you described hit me at about 2am after posting the question: "wait, why dont i just have an output from the first job, job2 'needs' job1, and add an if evaluating the output from job1?!" 🤦
